### PR TITLE
[MRG] Add more documentation, including a practical guide to parameters and preprocessing.

### PR DESCRIPTION
**Pull request by ctb**
Adds:

* more details on sourmash lca classify and sourmash lca summarize for @jiarong 
* a practical guide to running sourmash, including choice of k-mer size, scaled value, and preprocessing reads.
* known issues to more-info.md

Fixes #419 #238 #209 #254 #414.

Addresses part of #294.

Checklist:

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
Ready for review @brooksph @taylorreiter @halexand! This is an all-documentation PR :).

**Comment by ctb** [comment]
actually @jiarong could also review this :)

File: doc/using-sourmash-a-guide.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+### Computing signatures from a collection of genomes in a single file:
+
+```
+sourmash compute --scaled 1000 -k 21,31,51 file.fa --singleton
+```
+
+This computes signatures for all individual FASTA sequences in `file.fa`,
+names them based on their FASTA headers, and places them all in a single
+`.sig` file, `file.fa.sig`.  (This behavior is triggered by the option
```
Comment (jiarong) [inline doc/using-sourmash-a-guide.md]: `file.fa.sig` should be `FASTAHEADER.sig`?

File: doc/command-line.md [hunk 2]
```diff
@@ -247,38 +251,106 @@ Usage:
 sourmash lca --query query.sig [query2.sig ...] --db <lca db> [<lca db2> ...]
```
Comment (jiarong) [inline doc/command-line.md]: classify is left out:
sourmash lca classify --query query.sig [query2.sig ...] --db <lca db> [<lca db2> ...]

File: doc/command-line.md [hunk 3]
```diff
@@ -247,38 +251,106 @@ Usage:
...
 ```
 sourmash lca summarize --query query.sig [query2.sig ...] 
     --db <lca db> [<lca db2> ...]
 ```
 
-Example output:
+For example, the command line:
+
+```
+sourmash lca classify --query tests/test-data/63.fa.sig \
```
Comment (jiarong) [inline doc/command-line.md]: "classify" should be "summarize"

File: doc/command-line.md [hunk 4]
```diff
@@ -247,38 +251,106 @@ Usage:
...
+k-mers that have this classification; and the lineage classification.
+K-mer classifications are reported hierarchically, so the percentages
+and totals contain all assignments that are at a lower taxonomic level -
+e.g. *Bacteria*, above, contains all the k-mers in *Bacteria;Proteobacteria*.
+
+The same information is reported in a CSV file if `-o/--output` is used.
+
 ### `sourmash lca gather`
 
 The `sourmash lca gather` command finds all non-overlapping
```
Comment (jiarong) [inline doc/command-line.md]: "sourmash lca gather" is not discussed in classify-signatures.html yet.

**Comment by ctb** [comment]
Thanks for the detailed review @jiarong!  With respect to track abundance, what have you found useful? I don't really have a story for it :)

**Comment by jiarong** [comment]
In context of metagenome, the abundance info of each genome or taxon is generally important for diversity analysis (PCA, taxonomy distribution, and species richness etc.), e.g. for "sourmash gather", if we see the abundance of a species is > 50% of total in a soil metagenome, then it is probably a contaminant.
With track abundance, we can use median kmer abundance to estimate the abundance of each genome.

File: doc/using-sourmash-a-guide.md (hunk 1 above)
Comment (ctb) [reply doc/using-sourmash-a-guide.md]: no, the output filename is file.fa.sig based on file.fa.

File: doc/command-line.md (hunk 2 above)
Comment (ctb) [reply doc/command-line.md]: good catch!

File: doc/command-line.md (hunk 3 above)
Comment (ctb) [reply doc/command-line.md]: thank you!

**Comment by ctb** [comment]
Fixed in 9480873 and d9699a22! Ready for re-review! @jiarong

File: doc/classifying-signatures.md [hunk 5]
```diff
@@ -81,12 +81,24 @@ differences between the `sourmash lca` subcommands and the basic
...
 specific and if you get a match, it's pretty reliable. On the
 converse, however, k-mer identification is very brittle with respect
 to evolutionary divergence, so if you don't get a match it may only mean
 that the particular species isn't known.
 
+## Abundance weighting
+
+If you compute your input signatures with `--track-abundance`, both
+`sourmash gather` and `sourmash lca gather` will use that information
+to calculate an abundance-weighted result.  Briefly, this will weight
```
Comment (jiarong) [inline doc/classifying-signatures.md]: I think "median kmer multiplicity" should be a column in the output, so users can use that as an estimation of abundance of each genome in metaG. This is more of the feature request. It is all good as basic tutorial here.

**Pull request merged**